cmd/initializer/main.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"github.com/pingcap/tidb-operator/pkg/initializer"
+	"github.com/pingcap/tidb-operator/pkg/version"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/component-base/logs"
+	"k8s.io/klog"
+	"os"
+)
+
+var (
+	printVersion     bool
+	namespace        string
+	component        string
+	podName          string
+	verifyPeriodDays int
+)
+
+func init() {
+	flag.BoolVar(&printVersion, "V", false, "Show version and quit")
+	flag.BoolVar(&printVersion, "version", false, "Show version and quit")
+	flag.StringVar(&component, "component", "all", "The component to initializer resource for")
+	flag.IntVar(&verifyPeriodDays, "verifyPeriodDays", 365, "Number of days a certificate generated by -x509 is valid for.")
+	flag.BoolVar(&initializer.WebhookEnabled, "webhookEnabled", false, "switch to enable init webhook resource")
```

Comment:
As webhook server need to be enabled to use, Intializer need to switch on  `webhookEnabled` to initialize its resources.

Reply:
Webhook must be enabled when the component is `all` or `webhook`, so it seems like that this variable is redundant

Reply:
> Webhook must be enabled when the component is `all` or `webhook`, so it seems like that this variable is redundant

As webhook server is a `switch feature`, the webhook-server resources shouldn't be initialize when component is `all` and `webhookEnabled` is false which I think `webhookEnabled` is still necessary for now. It will be removed after Webhook enabled in default.

Reply:
Sounds like the problem is in the design of the `component` field, if the client (in this case, helm chart) choose to disable webhook initialization, then it could have excluded webhook in the component list, which is the only purpose of the `webhook-enabled` argument.

Reply:
> Sounds like the problem is in the design of the `component` argument, if the client (in this case, helm chart) choose to disable webhook initialization, then it could have excluded webhook in the component list, which is the only purpose of the `webhook-enabled` argument.

In this case, I intend to use `component` argument to decide which resource should be initialize. If we want to initializer all (possibly less then 5 or 3) the component resources except `webhook`,  the componenet should be defined as `aaa,bbb,ccc,ddd`. `webhook-enabled` should be removed after `webhook` enabled in default.